src/uart.rs

```diff
@@ -49,6 +49,7 @@ pub struct UartTx<'a, M: Mode> {
 /// Uart RX driver.
 pub struct UartRx<'a, M: Mode> {
     info: Info,
+    _buffer_config: Option<BufferConfig>,
```

Comment:
IMO, you should produce new structs that contain a `UartRx`. In fact, adding a new directory `uart` and move the current driver to `uart/mod.rs` would allow you to introduce all your new code within `uart/buffered.rs`. Then, you can have a new `struct BufferedUartRx`, `struct BufferedUartTx`, and `BufferedUart` all isolated from the regular, unbuffered uart. Additionally, you would need a `BufferConfig`. A regular `Config` would suffice, because you would already have a separate `buffered` module.